src/sys/options/options_ini.cxx

```diff
@@ -107,6 +106,39 @@ void OptionINI::read(Options *options, const string &filename) {
         string key, value;
         // Get a key = value pair
         parse(buffer, key, value);
+
+        // Ensure that brackets '()' and '[]'are balanced
+
+        // Count net number of opening and closing brackets
+        auto count_brackets = [](const string& input) {
```

Comment:
Another way of writing this, using std algorithms:

```cpp
auto count_brackets = [](const std::string& input) {
  return std::count_if(begin(input), end(input), [](const char& ch) { return ch == '(' or ch == '['; }) 
       - std::count_if(begin(input), end(input), [](const char& ch) { return ch == ')' or ch == ']'; });
}
```

Reply:
Thanks yes that would work. It traverses the string twice, but it is nicer in many ways. I'll change to this.